internal/registry/types.bucket.go

```diff
@@ -156,10 +156,10 @@ func (b *Bucket) UpdateBuildStatus(ctx context.Context, name string, status mode
 	_, err = b.client.UpdateBuild(ctx,
 		buildToUpdate.ID,
 		buildToUpdate.RunUUID,
-		buildToUpdate.CloudProvider,
```

Comment:
They were, just once they passed a certain point in the build (after the build is done, when post-processors start to be worked on), the status would never be updated again as all requests would fail.
Also a new run of the build would also never succeed to change the status as it would get rejected since the updates payload would contain the cloud provider, which gets rejected on HCP

Reply:
I'm not sure I understand the issue here. A build, if I'm not mistaken, will get the call to update the status to running once a build begins in the registry builder. The cloud provider information is derived from an build image (at the end of a build) so cloud provider will always be empty here. 

Unless a build was some how given a provider without an image. Is this possible on the API?

Reply:
I don't understand why the updating of build labels is being removed here. If the issue is that we can not update the cloud provider once set would it makes sense to consider the following options. 

1. Update the API to only error if the value of cloudprovider is being changed to some new value
2. Update the UpdateBuild  to check if the status of `buildToUpdate` is already set to DONE and if so bail as there is no need for a heartbeat. All builds are stored in memory and refreshed after a successful API call so that state in memory should contain all the information needed to determine if a heartbeat update needs to be sent.

Reply:
> because updates to the field are rejected and we include it all the time, we can't send anymore updates to the build later on.

If cloud provider is set that means that an image was uploaded at least in Packer core, which should mean that the build is DONE. Is that not true?

Reply:
> Update the API to only error if the value of cloudprovider is being changed to some new value.

The API already does that! Labels can be updated and new labels will be concatenated with what's already stored.